lib/src/rules/empty_statements.dart

```diff
@@ -69,8 +69,27 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
+  bool definesSemantics(EmptyStatement node) {
+    var parent = node.parent;
+    if (parent is! SwitchPatternCase) return false;
+
+    var statements = parent.statements;
+    var justEmpties = true;
+
+    var index = 0;
+    for (var (i, statement) in statements.indexed) {
```

Comment:
I believe that this works, but it seems a bit more complex than necessary. If the `node` isn't the last statement, then we can return `false`. If it is, then we can iterate over the previous statements and return `false` as soon as we find one that is not an empty statement. Otherwise we return `true`.

Your call whether to change it.